crates/edr_rpc_server/src/node.rs

```diff
@@ -13,15 +13,16 @@ use edr_eth::{
     },
     serde::ZeroXPrefixedBytes,
     signature::Signature,
-    Address, Bytes, U256, U64,
+    transaction::{EthTransactionRequest, SignedTransaction},
+    Address, Bytes, B256, U256, U64,
 };
 use edr_evm::{
     blockchain::BlockchainError,
     state::{AccountModifierFn, StateError},
     AccountInfo, Block, Bytecode, MineBlockResult, KECCAK_EMPTY,
 };
 
-use crate::{filter::Filter, node::node_data::NodeData, Config};
+use crate::{node::node_data::NodeData, Config};
 
 pub use self::node_error::NodeError;
 
```

Comment:
As the node is now a wrapper around `Mutex<NodeData>`, can't we remove one layer here? A lot of the functions seem like boilerplate and doing this would resolve it

Reply:
Yeah making the state `Arc<Mutex<NodeData>>` instead of `Arc<Node>` for the server would work, but I think we still have to validate if a single `Mutex` is ok performance-wise, so I'd rather keep all the locking related logic in one place, abstracted behind `Node`, in case we have to refactor later.

The pattern I'm trying to follow with what to put into `NodeData` is this: if the method is a helper method, then put it in `NodeData` since it'd try to acquire the lock twice resulting in a deadlock if it was in `Node` . But then there will be some methods like `new_pending_transaction_filter` in `NodeData` that are not helper methods, but are needed to test stuff in `NodeData`, so yeah that brings boilerplate.